sdk/appconfiguration/azure-data-appconfiguration/src/configuration_client.cpp

```diff
@@ -0,0 +1,2316 @@
+  url.AppendQueryParameter("label", Core::Url::Encode(options.Label));
+  url.AppendQueryParameter("After", Core::Url::Encode(options.After));
+  {
+    std::string valueStr;
+    for (size_t i = 0; i < options.Select.size(); ++i)
+    {
+      if (i != 0)
+      {
+        valueStr += ',';
+      }
+
+      valueStr += Core::Url::Encode(options.Select[i].ToString());
+    }
+
+    url.AppendQueryParameter("$Select", valueStr);
+  }
+
+  url.AppendQueryParameter("snapshot", Core::Url::Encode(options.Snapshot));
+  {
+    std::string valueStr;
+    for (size_t i = 0; i < options.Tags.size(); ++i)
+    {
+      if (i != 0)
+      {
+        valueStr += "&tags=";
+      }
+
+      valueStr += Core::Url::Encode(options.Tags[i]);
+    }
+
+    url.AppendQueryParameter("tags", valueStr);
+  }
+
+  Core::Http::Request request(Core::Http::HttpMethod::Head, url);
+
+  request.SetHeader("Accept", "application/json");
+
+  request.SetHeader("Sync-Token", options.SyncToken);
+  request.SetHeader("Accept-Datetime", options.AcceptDatetime);
+  request.SetHeader("If-Match", options.IfMatch);
+  request.SetHeader("If-None-Match", options.IfNoneMatch);
+  request.SetHeader("x-ms-client-request-id", options.XMsClientRequestId.ToString());
+  auto rawResponse = m_pipeline->Send(request, context);
+  auto const httpStatusCode = rawResponse->GetStatusCode();
+
+  if (httpStatusCode != Core::Http::HttpStatusCode::Ok)
+  {
+    throw Core::RequestFailedException(rawResponse);
+  }
+
+  CheckKeyValuesResult response{};
+  response.SyncToken = rawResponse->GetHeaders().at("Sync-Token");
+  response.ETag = rawResponse->GetHeaders().at("ETag");
+
+  return Response<CheckKeyValuesResult>(std::move(response), std::move(rawResponse));
+}
+
+Azure::Response<GetKeyValueResult> ConfigurationClient::GetKeyValue(
+    std::string const& key,
+    std::string const& accept,
+    GetKeyValueOptions const& options,
+    Core::Context const& context)
+{
+  auto url = m_url;
+  url.AppendPath("kv/");
+  url.AppendPath(!key.empty() ? Core::Url::Encode(key) : "null");
+
+  url.AppendQueryParameter("api-version", Core::Url::Encode(m_apiVersion));
+  url.AppendQueryParameter("label", Core::Url::Encode(options.Label));
+  {
+    std::string valueStr;
+    for (size_t i = 0; i < options.Select.size(); ++i)
+    {
+      if (i != 0)
+      {
+        valueStr += ',';
+      }
+
+      valueStr += Core::Url::Encode(options.Select[i].ToString());
+    }
+
+    url.AppendQueryParameter("$Select", valueStr);
+  }
+
+  Core::Http::Request request(Core::Http::HttpMethod::Get, url);
+
+  request.SetHeader("Accept", accept);
+  request.SetHeader("Sync-Token", options.SyncToken);
+  request.SetHeader("Accept-Datetime", options.AcceptDatetime);
+  request.SetHeader("If-Match", options.IfMatch);
+  request.SetHeader("If-None-Match", options.IfNoneMatch);
+  request.SetHeader("x-ms-client-request-id", options.XMsClientRequestId.ToString());
```

Comment:
You are right! If this header is set to an empty string, the RequestIdPolicy will find the header and not set it to a new Uuid. I am not sure if such an option needs to be exposed.

We don't expose any `Uuid` fields in options/models from the SDK, including client request id.

It isn't clear from the typespec definition where the `ClientRequestIdHeader` model is coming from and whether it should be optional or not, so needs some digging into to see if there is an emitter issue or a typespec issue. It might just be indicating we need to set this somehow, without exposing the ability to the customer to set it via options (and what RequestIdPolicy does might be sufficient).

https://github.com/Azure/azure-rest-api-specs/blob/7dfd17073198d5cf3df99a106b770cb01b395f2a/specification/appconfiguration/AppConfiguration/routes.tsp#L300-L306

Will follow-up with typespec folks here.

We have precedence in storage, where such IDs are `Nullable<std::string>` (within the protocol layer, but not exposed), so we may want to follow suite if we really need to:
https://github.com/Azure/azure-sdk-for-cpp/blob/3687136d04cc8e425cb368ce879a1686834b17c9/sdk/storage/azure-storage-files-datalake/src/rest_client.cpp#L56-L59

That said, it doesn't seem like anyone is setting these `.RequestId` options anywhere:
https://github.com/Azure/azure-sdk-for-cpp/pull/6184